src/App.js

```diff
@@ -1,25 +1,163 @@
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React, { Component } from 'react';
+import styles from './App.module.css';
+import DateRangePicker from './components/DateRangePicker/DateRangePicker';
+import CalendarYearly from './components/Calendar/CalendarYearly';
+import CalendarQuarterly from './components/Calendar/CalendarQuarterly';
+import CalendarMonthly from './components/Calendar/CalendarMonthly';
+import EventList from './components/EventList/EventList';
+const faker = require('faker');
+
+const generatorEventList = (size=3) => {
+  const eventList = Array(size).fill().map((_,index)=>{
+    return (
+      {id:index+1, name:faker.git.commitMessage(), location: faker.address.streetName(), hostname:faker.name.firstName(), type:faker.random.boolean()?"Private":"Public", date:faker.date.between('2018-01-01', '2022-01-05')}
+    );
+  });
+  eventList.sort((a, b) => (a.date > b.date) ? 1 : -1)
+  return eventList;
+}
+
+class App extends Component {
+  state ={
+    eventList: generatorEventList(12),
+    startDate: new Date("2018-1-1"),
+    endDate: new Date("2022-12-31"),
+    yearLowerLimit:2018,
+    yearAmmountLimit:5,
+    period:"Yearly",
+    selectAll:true,
+    year:2020,
+    month1:0,
+    month2:11
+  };
+
+  setStatePeriod = (val) => {
+    this.setState( { period: val } );
+  }
+
+  setStateSelectAll = (val)=>{
+    this.setState({selectAll : val});
+    if(val){
+      let firstDay = new Date(this.state.yearLowerLimit, 0, 1);
+      let lastDay = new Date(this.state.yearLowerLimit+this.state.yearAmmountLimit, 11, 31);
+      this.setState({startDate : firstDay});
+      this.setState({endDate : lastDay});
+    }
+  }
+
+  setStateYear = (val)=>{
+    this.setState({year : val});
+    let firstDay = this.state.startDate;
+    let lastDay = this.state.endDate;
+
+    firstDay = new Date(val, firstDay.getMonth(), 1);
+    lastDay = new Date(val, lastDay.getMonth()+1, 0);
+
+    this.setState({startDate : firstDay});
+    this.setState({endDate : lastDay});
+  }
+
+  setStateDateFromYear = (year) => {
+    let firstDay = new Date(year, 0, 1);
+    let lastDay = new Date(year, 11, 31);
+    this.setState({year : year});
+    this.setState({month1 : 0});
+    this.setState({month2 : 11});
+    this.setState({startDate : firstDay});
+    this.setState({endDate : lastDay});
+    this.setState({selectAll : false});
```

Comment:
@edwartexe be aware of spaces and code readability
```suggestion
    this.setState({ year: year });
    this.setState({ month1: 0 });
    this.setState({ month2: 11 });
    this.setState({ startDate: firstDay });
    this.setState({ endDate: lastDay });
    this.setState({ selectAll: false });
```